# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1449207825 -3600
#      Fri Dec 04 06:43:45 2015 +0100
# Node ID e870a68742a6c6e86333c72956aa31ceb7986131
# Parent  f7b0572fff6e5e555a3ecea2ee5c9bf8b6798910
change alignment of dock widget titles for OSX (bug #46592)

* octave-dock-widget.cc (set_style): change alignment in style sheet for OSX

diff --git a/libgui/src/octave-dock-widget.cc b/libgui/src/octave-dock-widget.cc
--- a/libgui/src/octave-dock-widget.cc
+++ b/libgui/src/octave-dock-widget.cc
@@ -325,16 +325,21 @@ octave_dock_widget::set_style (bool acti
 
   QString icon_col = _icon_color;
 
   if (_floating)
     dock_icon = "widget-dock";
   else
     dock_icon = "widget-undock";
 
+#if defined (Q_OS_MAC)
+  QString alignment = "center";
+#else
+  QString alignment = "center left";
+#endif
   if (_custom_style)
     {
 
       QColor bg_col, fg_col;
 
       if (active)
         {
           bg_col = _bg_color_active;
@@ -366,33 +371,33 @@ octave_dock_widget::set_style (bool acti
         arg (bg_col_top.name ()).
         arg (bg_col.name ()).
         arg (bg_col_bottom.name ());
 
 #if defined (Q_OS_WIN32)
       css = background + QString (" color: %1 ;").arg (fg_col.name ());
 #else
       css = QString ("QDockWidget::title { " + background +
-                     "                     text-align: center left;"
+                     "                     text-align: " + alignment + ";"
                      "                     padding: 0px 0px 0px 4px;}\n"
                      "QDockWidget { color: %1 ; "
                      "  titlebar-close-icon: url(:/actions/icons/widget-close%2.png);"
                      "  titlebar-normal-icon: url(:/actions/icons/"+dock_icon+"%2); }"
                      "QDockWidget::close-button,"
                      "QDockWidget::float-button { border: 0px; icon-size: %3px; width: %3px}"
                      ).
                      arg (fg_col.name ()).arg (icon_col).arg (_icon_size);
 #endif
     }
   else
     {
 #if defined (Q_OS_WIN32)
       css = QString ("");
 #else
-      css = QString ("QDockWidget::title { text-align: center left;"
+      css = QString ("QDockWidget::title { text-align: " + alignment + ";"
                      "                     padding: 0px 0px 0px 4px;}"
                      "QDockWidget {"
                      "  titlebar-close-icon: url(:/actions/icons/widget-close.png);"
                      "  titlebar-normal-icon: url(:/actions/icons/"+dock_icon+"); }"
                      "QDockWidget::close-button,"
                      "QDockWidget::float-button { border: 0px; icon-size: %1px; width: %1px}"
                     ).arg (_icon_size);
 #endif
